x/feegrant/internal/keeper/keeper.go

```diff
@@ -0,0 +1,157 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/tendermint/tendermint/libs/log"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/feegrant/exported"
+	"github.com/cosmos/cosmos-sdk/x/feegrant/internal/types"
+)
+
+// Keeper manages state of all fee grants, as well as calculating approval.
+// It must have a codec with all available allowances registered.
+type Keeper struct {
+	cdc      *codec.Codec
+	storeKey sdk.StoreKey
+}
+
+// NewKeeper creates a fee grant Keeper
+func NewKeeper(cdc *codec.Codec, storeKey sdk.StoreKey) Keeper {
+	return Keeper{cdc: cdc, storeKey: storeKey}
+}
+
+// Logger returns a module-specific logger.
+func (k Keeper) Logger(ctx sdk.Context) log.Logger {
+	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
+}
+
+// GrantFeeAllowance creates a new grant
+func (k Keeper) GrantFeeAllowance(ctx sdk.Context, grant types.FeeAllowanceGrant) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.FeeAllowanceKey(grant.Granter, grant.Grantee)
+	bz := k.cdc.MustMarshalBinaryBare(grant)
+	store.Set(key, bz)
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeSetFeeGrant,
+			sdk.NewAttribute(types.AttributeKeyGranter, grant.Granter.String()),
+			sdk.NewAttribute(types.AttributeKeyGrantee, grant.Grantee.String()),
+		),
+	)
+}
+
+// RevokeFeeAllowance removes an existing grant
+func (k Keeper) RevokeFeeAllowance(ctx sdk.Context, granter, grantee sdk.AccAddress) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.FeeAllowanceKey(granter, grantee)
+	store.Delete(key)
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeRevokeFeeGrant,
+			sdk.NewAttribute(types.AttributeKeyGranter, granter.String()),
+			sdk.NewAttribute(types.AttributeKeyGrantee, grantee.String()),
+		),
+	)
+}
+
+// GetFeeAllowance returns the allowance between the granter and grantee.
+// If there is none, it returns nil, nil.
+// Returns an error on parsing issues
+func (k Keeper) GetFeeAllowance(ctx sdk.Context, granter, grantee sdk.AccAddress) exported.FeeAllowance {
+	grant, found := k.GetFeeGrant(ctx, granter, grantee)
+	if !found {
+		return nil
+	}
+	return grant.Allowance
+}
+
+// GetFeeGrant returns entire grant between both accounts
+func (k Keeper) GetFeeGrant(ctx sdk.Context, granter sdk.AccAddress, grantee sdk.AccAddress) (types.FeeAllowanceGrant, bool) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.FeeAllowanceKey(granter, grantee)
```

Comment:
ref: https://github.com/cosmos/cosmos-sdk/issues/5254

We should ideally be using `prefix.NewStore`. This is the design approach going forward with new modules. See the evidence module for guidance.

Reply:
Yes. It is very hard to pass a review with a moving target and new standards being added between every review pass. This seemed to be standard when I started the PR. I can update.